Python/fileutils.c

```diff
@@ -2771,6 +2771,10 @@ _Py_get_osfhandle(int fd)
 int
 _Py_open_osfhandle_noraise(void *handle, int flags)
 {
+    // PEP 446: Create non inheritable file descriptor by default,
+    // always set the _O_NOINHERIT flag.
+    flags |= _O_NOINHERIT;
```

Comment:
Why not let the callers pass this value in? They should know whether the handle is inheritable or not

Reply:
> Are there cases when Python creates inheritable handles on purpose?

Not yet, but I could see multiprocessing wanting to do it, and choosing to use an existing wrapper for a C runtime API instead of calling it directly. It'll save someone in the future some debugging.

> Do you expect msvcrt.open_osfhandle() to return a inheritable FD if the handle is inheritable?

Only if the caller doesn't pass `O_NOINHERIT`. It doesn't have to figure out whether the handle is inheritable or not, it just has to trust the flags that are passed.

Right now, it seems to be making it inheritable because the flag isn't being passed, so presumably that would actually work fine if the underlying handle really were inheritable. That's a simple boolean (you can see it in the `os.pipe` implementation, set to `FALSE`), so no reason it couldn't be done.

Reply:
Instead of modifying _Py_get_osfhandle(), I modified _Py_get_osfhandle() callers to pass _O_NOINHERIT.